pkg/blockbuilder/blockbuilder.go

```diff
@@ -239,3 +237,306 @@ func (b *BlockBuilder) getLagForPartition(ctx context.Context, partition int32)
 
 	return kadm.GroupMemberLag{}, lastErr
 }
+
+type partitionState struct {
+	// Lag is the upperbound number of records the cycle will need to consume.
+	Lag int64
+	// Commit is the current offset commit for the partition.
```

Comment:
[nit] I suggest to clarify it.

```suggestion
	// Commit is the offset of the next record we'll start consuming.
```